WooCommerce/WooCommerceUITests/Tests/StatsTests.swift

```diff
@@ -24,4 +24,42 @@ final class StatsTests: XCTestCase {
             .goToThisYearTab()
             .verifyThisYearStatsLoaded()
     }
+
+    func test_view_detailed_chart_stats() throws {
+        var hourlyRevenue = ""
+        var dailyRevenue = ""
+        var weeklyRevenue = ""
+        var monthlyRevenue = ""
+        var yearlyRevenue = ""
+
+        try TabNavComponent()
+            .goToMyStoreScreen()
+
+        dailyRevenue = try MyStoreScreen().getRevenueValue()
+        try MyStoreScreen().tapChart()
+        hourlyRevenue = try MyStoreScreen().getRevenueValue()
+
+        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: dailyRevenue, updatedRevenue: hourlyRevenue)
+        try MyStoreScreen().goToThisWeekTab()
+
+        weeklyRevenue = try MyStoreScreen().getRevenueValue()
+        try MyStoreScreen().tapChart()
+        dailyRevenue = try MyStoreScreen().getRevenueValue()
+
+        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: weeklyRevenue, updatedRevenue: dailyRevenue)
+        try MyStoreScreen().goToThisMonthTab()
+
+        monthlyRevenue = try MyStoreScreen().getRevenueValue()
+        try MyStoreScreen().tapChart()
+        weeklyRevenue = try MyStoreScreen().getRevenueValue()
+
+        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: monthlyRevenue, updatedRevenue: weeklyRevenue)
+        try MyStoreScreen().goToThisYearTab()
+
+        yearlyRevenue = try MyStoreScreen().getRevenueValue()
+        try MyStoreScreen().tapChart()
+        monthlyRevenue = try MyStoreScreen().getRevenueValue()
+
+        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: yearlyRevenue, updatedRevenue: monthlyRevenue)
+    }
```

Comment:
A few things came to my mind while I was reviewing:

1. We are using mocks, so we always know the tested data. Why not assert for exact values, instead of checking that the value has changed?
2. If ^ is implemented, this test can use method chaining, which IMO makes it more readable and also shorter. I also believe the test will be faster, because `MyStoreScreen` won't be initialized multiple times (dummy code):

```suggestion

    func test_view_detailed_chart_stats() throws {
        try TabNavComponent()
            .goToMyStoreScreen()            
            // Checks for "Today"
            .assertRevenueValue(dailyRevenueExpected)
            .tapChart()
            .assertRevenueValue(hourlyRevenueExpected)            
            // Checks for "This Week"
            .goToThisWeekTab()
            .assertRevenueValue(weeklyRevenueExpected)
            .tapChart()
            .assertRevenueValue(dailyRevenueExpected)
            // Checks for "This Month"
            .goToThisMonthTab()
            .assertRevenueValue(monthlyRevenueExpected)
            .tapChart()
            .assertRevenueValue(weeklyRevenueExpected)
            // Checks for "This Year"
            .goToThisYearTab()
            .assertRevenueValue(yearlyRevenueExpected)
            .tapChart()
            .assertRevenueValue(monthlyRevenueExpected)
    }
```

If you decide to go this way: I'm not sure if it's better to define all expected values as variables at the test start, or just to feed the raw values to the function. It depends on if the same value stays the same between tabs.

3. If you decide not to implement `1` and `2`, I think the test would benefit from this cosmetic restructuring, making it easier to see the separate logical blocks (I also assigned the value to `dailyRevenue` in a bit different way, causing one less init of `MyStoreScreen()`:

```suggestion

    func test_view_detailed_chart_stats() throws {
        var hourlyRevenue = ""
        var dailyRevenue = ""
        var weeklyRevenue = ""
        var monthlyRevenue = ""
        var yearlyRevenue = ""

        dailyRevenue = try TabNavComponent()
            .goToMyStoreScreen()
            .getRevenueValue()
        try MyStoreScreen().tapChart()
        hourlyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: dailyRevenue, updatedRevenue: hourlyRevenue)

        try MyStoreScreen().goToThisWeekTab()
        weeklyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().tapChart()
        dailyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: weeklyRevenue, updatedRevenue: dailyRevenue)

        try MyStoreScreen().goToThisMonthTab()
        monthlyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().tapChart()
        weeklyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: monthlyRevenue, updatedRevenue: weeklyRevenue)

        try MyStoreScreen().goToThisYearTab()
        yearlyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().tapChart()
        monthlyRevenue = try MyStoreScreen().getRevenueValue()
        try MyStoreScreen().verifyRevenueUpdated(originalRevenue: yearlyRevenue, updatedRevenue: monthlyRevenue)
    }
```

Reply:
> We are using mocks, so we always know the tested data. Why not assert for exact values, instead of checking that the value has changed?
the reason is that when the test is tapping on the chart using `tapChart()` wasn't sure if it will tap on the same exact point each time so asserting for exact values might end up returning flaky results. I'll test a bit to see if this can work on different devices.

Reply:
i tested the image chart tap on multiple devices and it seems like at this moment it does tap on the same coordinates on the screen, though I'm still a bit uncomfortable with the idea of depending on that click (since we're not clicking on a specific coordinate it can change if say the design of the chart change). 

but i totally get what you mean that it's a bit hard to read the way it's structured now and the test is a bit slow, so I've updated it following your suggestion in point 3 with some additional changes, the change cut about ~15 seconds of runtime from the test. changes made in https://github.com/woocommerce/woocommerce-ios/pull/8319/commits/6d022f551e984cade6b8ec2abc65aa357617fdbc